cli/src/commands/enrich/snyk.rs

```diff
@@ -26,98 +26,74 @@ pub struct SnykArgs {
-        let storage: Box<dyn StorageProvider>;
+/// Concrete implementation of the command handler. Responsible for
+/// dispatching command to the correct logic handler based on args passed.
+pub async fn execute(args: &EnrichArgs) -> Result<(), Error> {
+    let storage: Box<dyn StorageProvider>;
+    let token = harbcore::config::snyk_token().map_err(|e| Error::Config(e.to_string()))?;
 
-        let cx = match &args.debug {
-            false => {
-                storage = Box::new(S3StorageProvider {});
-                harbcore::config::harbor_context().map_err(|e| Error::Config(e.to_string()))?
-            }
-            true => {
-                storage = Box::new(FileSystemStorageProvider::new(
-                    "/tmp/harbor-debug/vulnerabilities".to_string(),
-                ));
-                harbcore::config::dev_context(None).map_err(|e| Error::Config(e.to_string()))?
-            }
-        };
-
-        match &args.snyk_args {
-            None => {
-                let mut task: Task =
-                    Task::new(TaskKind::Vulnerabilities(VulnerabilityProviderKind::Snyk))
-                        .map_err(|e| Error::Enrich(e.to_string()))?;
-
-                let provider = SnykProvider::new_provider(cx, storage)
-                    .await
+    let cx = match &args.debug {
+        false => {
+            storage = Box::new(S3StorageProvider {});
+            harbcore::config::harbor_context().map_err(|e| Error::Config(e.to_string()))?
+        }
+        true => {
+            storage = Box::new(FileSystemStorageProvider::new(
+                "/tmp/harbor-debug/vulnerabilities".to_string(),
+            ));
+            harbcore::config::dev_context(None).map_err(|e| Error::Config(e.to_string()))?
+        }
+    };
+
+    let store = Arc::new(
+        Store::new(&cx)
+            .await
+            .map_err(|e| Error::Enrich(e.to_string()))?,
+    );
+
+    let provider = SyncTask::new(
+        store.clone(),
+        SnykService::new(token),
+        PackageService::new(store.clone()),
+        VulnerabilityService::new(store, storage),
+    )
+    .map_err(|e| Error::Enrich(e.to_string()))?;
+
+    match &args.snyk_args {
+        None => {
+            let mut task: Task =
+                Task::new(TaskKind::Vulnerabilities(VulnerabilityProviderKind::Snyk))
                     .map_err(|e| Error::Enrich(e.to_string()))?;
 
-                provider
-                    .execute(&mut task)
-                    .await
-                    .map_err(|e| Error::Enrich(e.to_string()))
-            }
-            Some(_a) => Err(Error::Enrich(
-                "individual project not yet implemented".to_string(),
-            )),
+            provider
+                .execute(&mut task)
+                .await
+                .map_err(|e| Error::Enrich(e.to_string()))
         }
+        Some(_a) => Err(Error::Enrich(
+            "individual project not yet implemented".to_string(),
```

Comment:
Is the state of this documented anywhere?

Reply:
No, but stay tuned. 😉